### PR TITLE
Fixes #YUPTOO-37 - omit nics starts with & associated mac_addrs

### PR DESCRIPTION
**Pull request by kgaikwad**

### CODE REVIEW

File: yuptoo/modifiers/transform_mac_addresses.py [hunk 1]
```diff
@@ -1,7 +1,26 @@
...
+    host: dict, mac_addresses_to_omit: list,
+        transformed_obj=copy.deepcopy(TRANSFORMED_DICT)):
+    """Remove mac_addresses for omitted nics."""
+
+    mac_addresses = host.get('mac_addresses')
+    len_of_mac_addrs_to_omit = len(mac_addresses_to_omit)
+    if mac_addresses and len_of_mac_addrs_to_omit > 0:
+        host['mac_addresses'] = list(
+            set(mac_addresses) - set(mac_addresses_to_omit))
+        if not host['mac_addresses']:
```
Comment (r14chandra) [inline yuptoo/modifiers/transform_mac_addresses.py:19]: `and` operator (line 16) already covers `!host[''mac_addresses]`. So, the statement under this `if` (line 19) won't be executed anyway. Is my assumption correct? Can we align it with the `if` statement (at line 16)?

**Comment by patilsuraj767** [comment]
Hello @kgaikwad Thanks for creating the PR, Modifiers are designed to be independent of each other. One modifier should not import the other. Will request you to shift whole code under single modifier.

File: yuptoo/modifiers/transform_mac_addresses.py (hunk 1 above)
Comment (kgaikwad) [reply yuptoo/modifiers/transform_mac_addresses.py:19]: Intention behind this line is when there is empty list as result of difference between host's current mac_addresses values & mac_addresses_to_omit.

**Comment by kgaikwad** [comment]
> Hello @kgaikwad Thanks for creating the PR, Modifiers are designed to be independent of each other. One modifier should not import the other. Will request you to shift whole code under single modifier.

As I am modifying `mac_addresses` field details of host, I won't prefer to move `mac_addresses` specific code into file named as `transform_network_interfaces.py`. In addition to this, logic extracted into a static method so it can be easily imported from anywhere.
My intention was to keep all transformations related to mac_addresses inside one file i.e `transform_mac_addresses.py` so that in future one can add changes just by looking into single file and anticipate the impact of change.
I can move this method to different utils file, will it work for you? thoughts?

File: yuptoo/modifiers/transform_mac_addresses.py (hunk 1 above)
Comment (r14chandra) [reply yuptoo/modifiers/transform_mac_addresses.py:19]: Got it 👍🏻 Thanks!

**Comment by patilsuraj767** [comment]
Yes, we can move the logic in different utils file which is outside modifier directory. Once [issue 30](https://github.com/RedHatInsights/yuptoo/issues/30) is fixed we can move the code to new modifier file.

**Comment by kgaikwad** [comment]
Moved helper method to different file. Please request to review.

**Comment by kgaikwad** [comment]
> Once [issue 30](https://github.com/RedHatInsights/yuptoo/issues/30) is fixed we can move the code to new modifier file.

Btw, I didn't get how this is relevant to [issue 30](https://github.com/RedHatInsights/yuptoo/issues/30). As per understanding these are two different things.

**Review by r14chandra — APPROVED** [review]
lgtm! 👍🏻

**Comment by patilsuraj767** [comment]
> > Once [issue 30](https://github.com/RedHatInsights/yuptoo/issues/30) is fixed we can move the code to new modifier file.
> 
> Btw, I didn't get how this is relevant to [issue 30](https://github.com/RedHatInsights/yuptoo/issues/30). As per understanding these are two different things.

@kgaikwad what I meant was once we have issue 30 fixed and once we have dependency graph mechanism in place. Then we can create a separate modifier for omitting nics and add its dependency in transform_network_interfaces.py. That way all the code can be in one file + transform_network_interfaces modifier will run after omitting nics modifier.

**Review by patilsuraj767 — APPROVED** [review]
Thanks for the PR @kgaikwad, Approving and merging it.

**Pull request merged**